testing/test_assertrewrite.py

```diff
@@ -135,26 +135,26 @@ def test_rewrites_plugin_as_a_package(self, testdir):
             "        assert x == y\n"
             "    return special_assert\n"
         )
-        testdir.makeconftest('pytest_plugins = ["plugin"]')
-        testdir.makepyfile("def test(special_asserter): special_asserter(1, 2)\n")
-        result = testdir.runpytest()
+        pytester.makeconftest('pytest_plugins = ["plugin"]')
+        pytester.makepyfile("def test(special_asserter): special_asserter(1, 2)\n")
+        result = pytester.runpytest()
         result.stdout.fnmatch_lines(["*assert 1 == 2*"])
 
-    def test_honors_pep_235(self, testdir, monkeypatch):
+    def test_honors_pep_235(self, pytester, monkeypatch):
         # note: couldn't make it fail on macos with a single `sys.path` entry
         # note: these modules are named `test_*` to trigger rewriting
-        testdir.tmpdir.join("test_y.py").write("x = 1")
-        xdir = testdir.tmpdir.join("x").ensure_dir()
+        pytester.tmpdir.join("test_y.py").write("x = 1")
```

Comment:
With `pytester`, `testdir.tmpdir` becomes `pytester.path`. It also becomes a `pathlib.Path`, so the code would need to be a little different. In this case:

```suggestion
        pytester.tmpdir.joinpath("test_y.py").write_text("x = 1", "utf-8")
```